sdk/identity/Azure.Identity/src/VisualStudioCredential.cs

```diff
@@ -0,0 +1,242 @@
+            }
+        }
+
+        private async Task<AccessToken> RunProcessesAsync(List<ProcessStartInfo> processStartInfos, bool async, CancellationToken cancellationToken)
+        {
+            var exceptions = new List<Exception>();
+            foreach (ProcessStartInfo processStartInfo in processStartInfos)
+            {
+                try
+                {
+                    var processRunner = new ProcessRunner(_processService.Create(processStartInfo), TimeSpan.FromSeconds(30), cancellationToken);
+                    string output = async
+                        ? await processRunner.RunAsync().ConfigureAwait(false)
+                        : processRunner.Run();
+
+                    JsonElement root = JsonDocument.Parse(output).RootElement;
+                    string accessToken = root.GetProperty("access_token").GetString();
+                    DateTimeOffset expiresOn = root.GetProperty("expires_on").GetDateTimeOffset();
+                    return new AccessToken(accessToken, expiresOn);
+                }
+                catch (Exception exception)
+                {
+                    exceptions.Add(exception);
+                }
+            }
+
+            switch (exceptions.Count) {
+                case 0:
+                    throw new InvalidOperationException();
+                case 1:
+                    ExceptionDispatchInfo.Capture(exceptions[0]).Throw();
+                    return default;
+                default:
+                    throw new AggregateException(exceptions);
+            }
+        }
+
+        private List<ProcessStartInfo> GetProcessStartInfos(VisualStudioTokenProvider[] visualStudioTokenProviders, string resource, CancellationToken cancellationToken)
+        {
+            List<ProcessStartInfo> processStartInfos = new List<ProcessStartInfo>();
+            StringBuilder arguments = new StringBuilder();
+
+            foreach (VisualStudioTokenProvider tokenProvider in visualStudioTokenProviders)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                // If file does not exist, the version of Visual Studio that set the token provider may be uninstalled.
+                if (!_fileSystem.FileExists(tokenProvider.Path))
+                {
+                    continue;
+                }
+
+                arguments.Clear();
+                arguments.Append(ResourceArgumentName).Append(' ').Append(resource);
+
+                if (_tenantId != default)
+                {
+                    arguments.Append(' ').Append(TenantArgumentName).Append(' ').Append(_tenantId);
+                }
+
+                // Add the arguments set in the token provider file.
+                if (tokenProvider.Arguments?.Length > 0)
+                {
+                    foreach (var argument in tokenProvider.Arguments)
+                    {
+                        arguments.Append(' ').Append(argument);
```

Comment:
can arguments contain spaces? Do we need to quote them?

Reply:
They already come quoted, e.g.:
```
["--serviceHubConfig","\"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\Common7\\servicehub.config.json\""]
```